### PR TITLE
Refactor MarkdownChangelogEntry

### PR DESCRIPTION
**Pull request by luqkrzy**
- refactor MarkdownChangelogEntry into subclasses int 'entry' package
- add tests to 'entry' package classes

### CODE REVIEW

File: src/test/java/com/github/marwin1991/keep_changelog/markdown/MarkdownChangelogEntryTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                .mergeRequest("Merge Request")
+                .issues(List.of("issue1", "issue2", "issue3"))
+                .links(List.of(new ChangelogLink("google", "google.com"), new ChangelogLink("wp", "wp.pl")))
+                .type(ChangeLogEntryType.ADDED)
+                .importantNotes(List.of("Important one", "Important two"))
+                .configuration(List.of(new Configuration("conf type", ConfigurationAction.ADD, "key", "def value", "some description", "more info here")))
+                .build();
+
+        // when
+        // TODO Is that correct output for markdownChangelogEntry ?
```
Comment (marwin1991) [inline src/test/java/com/github/marwin1991/keep_changelog/markdown/MarkdownChangelogEntryTest.java:28]: yes

Comment (marwin1991) [reply src/test/java/com/github/marwin1991/keep_changelog/markdown/MarkdownChangelogEntryTest.java:28]: Altho  mergeRequest can be a number, like !50

File: src/main/java/com/github/marwin1991/keep_changelog/markdown/entry/MarkdownChangelogEntryAuthor.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        return StringUtils.defaultString(author.getName());
+    }
+
+    private String getAuthorNick(Author author) {
+        if (StringUtils.isBlank(author.getNick())) {
+            return StringUtils.EMPTY;
+        }
+
+        return "@" + author.getNick();
```
Comment (marwin1991) [inline src/main/java/com/github/marwin1991/keep_changelog/markdown/entry/MarkdownChangelogEntryAuthor.java:57]: "@" maybe as constant field?

**Review by marwin1991 — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/marwin1991/keep_changelog/markdown/entry/MarkdownChangelogEntryAuthor.java (hunk 2 above)
Comment (luqkrzy) [reply src/main/java/com/github/marwin1991/keep_changelog/markdown/entry/MarkdownChangelogEntryAuthor.java:57]: OK

File: pom.xml [hunk 3]
```diff
@@ -178,6 +178,14 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <version>2.22.2</version>
             </plugin>
+            <plugin>
```
Comment (marwin1991) [inline pom.xml:181]: I think this part is not needed. 
This plugin is build with java 8

Comment (luqkrzy) [reply pom.xml]: ok, fixed

**Comment by marwin1991** [comment]
Really well done :)

**Review by marwin1991 — APPROVED** [review]

**Pull request merged**